pkg/p2p/server.go

```diff
@@ -0,0 +1,834 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package p2p
+
+import (
+	"context"
+	"reflect"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/workerpool"
+	"github.com/pingcap/ticdc/proto/p2p"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc/codes"
+	gRPCPeer "google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+// MessageServerConfig stores configurations for the MessageServer
+type MessageServerConfig struct {
+	// The maximum number of entries to be cached for topics with no handler registered
+	MaxTopicPendingCount int
+	// The maximum number of unhandled internal tasks for the main thread.
+	MaxPendingTaskCount int
```

Comment:
Do we have to unify the naming convention of `MaxTopicPendingCount` and `MaxPendingTaskCount`.

How about rename `MaxPendingTaskCount` to `MaxTaskPendingCount`